nifi-extension-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/GetSlackReaction.java

```diff
@@ -0,0 +1,294 @@
+            .build();
+
+    static final PropertyDescriptor THREAD_TIMESTAMP = new PropertyDescriptor.Builder()
+            .name("Message Timestamp")
+            .description("Slack message's timestamp the reactions are fetched for.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    static final PropertyDescriptor RELEASE_IF_ONE_REACTION = new PropertyDescriptor.Builder()
+            .name("Release If One Reaction Arrived")
+            .description("It is possible to wait for the first reaction or wait for the specified amount of time to fetch all possible reactions.")
+            .required(true)
+            .allowableValues("true", "false")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .defaultValue("true")
+            .build();
+
+    static final PropertyDescriptor WAIT_MONITOR_WINDOW = new PropertyDescriptor.Builder()
+            .name("Wait Monitor Window")
+            .description("Processor will periodically poll reactions for the given time duration if no reaction arrived yet or " +
+                    RELEASE_IF_ONE_REACTION.getDisplayName() + " is set to false.")
+            .required(true)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .defaultValue("1 hour")
+            .build();
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("FlowFiles are routed to this relationship if fetching reactions were successful")
+            .build();
+
+    static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("FlowFiles are routed to this relationship if an error occurred when fetching reactions")
+            .build();
+
+    static final Relationship REL_WAIT = new Relationship.Builder()
+            .name("wait")
+            .description("Self loop relationship which is used when waiting for further reactions")
+            .build();
+
+    static final Relationship REL_NO_REACTION = new Relationship.Builder()
+            .name("no_reaction")
+            .description("FlowFiles are routed to this relationship if no reaction arrived for the message in the given timeframe")
+            .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return Arrays.asList(CHANNEL_ID,
+                             ACCESS_TOKEN,
+                             THREAD_TIMESTAMP,
+                             WAIT_MONITOR_WINDOW,
+                             RELEASE_IF_ONE_REACTION);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return Set.of(REL_SUCCESS,
+                REL_WAIT,
+                REL_NO_REACTION,
+                REL_FAILURE);
+    }
```

Comment:
From what I have seen in the code base its best to define the collections for descriptors and relationships in static final variable and return them in the corresponding get methods.

```suggestion
 private static final List<PropertyDescriptor> DESCRIPTORS = List.of(CHANNEL_ID,
                             ACCESS_TOKEN,
                             THREAD_TIMESTAMP,
                             WAIT_MONITOR_WINDOW,
                             RELEASE_IF_ONE_REACTION);

 private static final Set<Relationship> RELATIONSHIPS = Set.of(REL_SUCCESS,
                REL_WAIT,
                REL_NO_REACTION,
                REL_FAILURE);
                
    @Override
    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
        return DESCRIPTORS;
    }

    @Override
    public Set<Relationship> getRelationships() {
        return RELATIONSHIPS;
    }
```